common/web/types/src/kmx/kmx-plus/kmx-plus.ts

```diff
@@ -135,6 +135,18 @@ export class StrsItem {
   isEqual(a: StrsItem): boolean {
     return a.value === this.value && a.char === this.char;
   }
+
+  private _context: any = undefined;
+
+  /** add any context from the options to this strsitem */
+  setContext(opts?: StrsOptions) {
+    // At present, there's only a single piece of context available
+    this._context = this._context || opts?.x;
+  }
+
+  get context() : any {
+    return this._context;
+  }
```

Comment:
Can we use consistent pattern here? (either both `get context()`/`set context()`, or `getContext()/setContext()`? (I am not a big fan of getters/setters because they hide behaviour for what looks like a simple assignment, but that's a separate story)